src/librustdoc/html/highlight.rs

```diff
@@ -229,7 +259,7 @@ fn get_real_ident_class(text: &str, edition: Edition, allow_path_keywords: bool)
     Some(match text {
         "ref" | "mut" => Class::RefKeyWord,
         "false" | "true" => Class::Bool,
-        _ if Symbol::intern(text).is_reserved(|| edition) => Class::KeyWord,
+        _ if Symbol::intern(text).is_reserved(|| Edition::Edition2021) => Class::KeyWord,
```

Comment:
This is quite a big change. Until now the highlight depended on the edition provided to rustdoc. Even though it is a welcomed simplification, I'm not too sure if it's a good idea...

Another problem I have with this approach: we will need to not forget to update it whenever the edition is updated. To go around that limitation, you could add a `current` method on `Edition` which would return the current highest edition. To be discussed I suppose.

Reply:
Also, it seems it's not taking into account the `edition` parameter that the users can use on codeblocks.